components/brave_wallet/browser/cardano/cardano_wallet_service.cc

```diff
@@ -97,12 +136,116 @@ void CardanoWalletService::OnGetCardanoUtxosTaskDone(
   auto cb = std::move(it->second);
   get_cardano_utxo_tasks_.erase(it);
 
-  if (!result.has_value()) {
-    std::move(cb).Run(nullptr, result.error());
+  std::move(cb).Run(std::move(result));
+}
+
+void CardanoWalletService::CreateCardanoTransaction(
+    mojom::AccountIdPtr account_id,
+    const CardanoAddress& address_to,
+    uint64_t amount,
+    bool sending_max_amount,
+    CardanoCreateTransactionTaskCallback callback) {
+  CHECK(IsCardanoAccount(account_id));
+
+  auto& task = create_transaction_tasks_.emplace_back(
+      std::make_unique<CardanoCreateTransactionTask>(
+          *this, account_id, address_to, amount, sending_max_amount),
+      std::move(callback));
+
+  task.first->Start(
+      base::BindOnce(&CardanoWalletService::OnCreateCardanoTransactionTaskDone,
+                     weak_ptr_factory_.GetWeakPtr(), task.first.get()));
+}
+
+void CardanoWalletService::OnCreateCardanoTransactionTaskDone(
+    CardanoCreateTransactionTask* task,
+    base::expected<CardanoTransaction, std::string> result) {
+  auto it = std::ranges::find(create_transaction_tasks_, task,
+                              [](auto& t) { return t.first.get(); });
```

Comment:
This is not needed with `base::UniquePtrComparator`

Reply:
removed